infrastructure/SharedInfrastructure/src/SharedInfrastructure/EC2ComputeStack.cs

```diff
@@ -0,0 +1,299 @@
+        Ec2Role = new Role(this, "EC2Role", new RoleProps
+        {
+            AssumedBy = new CompositePrincipal(new ServicePrincipal("ec2.amazonaws.com")),
+            ManagedPolicies = new[]
+            {
+                ManagedPolicy.FromAwsManagedPolicyName("AmazonSSMManagedInstanceCore"),
+                ManagedPolicy.FromAwsManagedPolicyName("service-role/AWSCodeDeployRole")
+            }
+        });
+
+        // Access to read parameters by path is not in the AmazonSSMManagedInstanceCore
+        // managed policy
+        Ec2Role.AddToPolicy(new PolicyStatement(new PolicyStatementProps
+        {
+            Effect = Effect.ALLOW,
+            Actions = new[] { "ssm:GetParametersByPath" },
+            Resources = new[]
+            {
+                Arn.Format(new ArnComponents
+                {
+                    Service = "ssm",
+                    Resource = "parameter",
+                    ResourceName = $"{Constants.AppName}/*"
+                }, this)
+            }
+        }));
+        
+        // Provide permission to allow access to Amazon Rekognition for processing uploaded
+        // book images. Credentials for the calls will be provided courtesy of the application
+        // role defined above, and the trust relationship with EC2.
+        Ec2Role.AddToPolicy(new PolicyStatement(new PolicyStatementProps
+        {
+            Effect = Effect.ALLOW,
+            Actions = new[] { "rekognition:DetectModerationLabels" },
+            Resources = new[] { "*" }
+        }));
+
+        // Add permissions for the app to retrieve the database password in Secrets Manager
+        Ec2Role.AddToPolicy(new PolicyStatement(new PolicyStatementProps
+        {
+            Effect = Effect.ALLOW,
+            Actions = new[]
+            {
+                "secretsmanager:DescribeSecret",
+                "secretsmanager:GetSecretValue"
+            },
+            Resources = new[]
+            {
+                "*"
+            }
+        }));
+
+        // Add permissions to the app to access the S3 image bucket
+        props.ImageBucket.GrantReadWrite(Ec2Role);
+
+        // Create an Amazon CloudWatch log group for the website
+        _ = new LogGroup(this, $"{Constants.AppName}LogGroup", new LogGroupProps
+        {
+            LogGroupName = Constants.AppName,
+            RemovalPolicy = RemovalPolicy.DESTROY
+        });
+
+        // Add permissions to write logs
+        Ec2Role.AddToPolicy(new PolicyStatement(new PolicyStatementProps
+        {
+            Effect = Effect.ALLOW,
+            Actions = new[]
+            {
+                "logs:DescribeLogGroups",
+                "logs:CreateLogGroup",
+                "logs:CreateLogStream"
+            },
+            Resources = new[]
+            {
+                "arn:aws:logs:*:*:log-group:*"
+            }
+        }));
+
+        // TODO Can this be combined with the PolicyStatement above?
```

Comment:
Did we already check and confirm it can't, or is this a future task?

Reply:
Looks good. I have combined the two policy statements.